src/Workspaces/Core/Portable/Execution/CustomAssetBuilder.cs

```diff
@@ -39,7 +41,9 @@ public CustomAsset Build(OptionSet options, string language, CancellationToken c
 
         public CustomAsset Build(AnalyzerReference reference, CancellationToken cancellationToken)
         {
-            return new WorkspaceAnalyzerReferenceAsset(reference, _serializer);
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
❗️ This is the wrong place to handle cancellation. It should be passed to the `WorkspaceAnalyzerReferenceAsset` constructor, and used there instead of the current use of `CancellationToken.None`.

Reply:
sure.